pkg/compliance/agent/telemetry.go

```diff
@@ -20,8 +20,8 @@ const (
 
 // telemetry reports environment information (e.g containers running) when the compliance component is running
 type telemetry struct {
-	sender   aggregator.Sender
-	detector collectors.DetectorInterface
+	sender    aggregator.Sender
+	ctrLister generic.ContainerAccessor
```

Comment:
any reason why we don't call [workloadmeta.ListContainers](https://github.com/DataDog/datadog-agent/blob/main/pkg/workloadmeta/store.go#L275 ) directly? Or maybe move the `ContainerAccessor` interface away from the `collector/corechecks/containers/generic` pkg? I find it a bit unusual that we're importing it from a corecheck pkg.

Reply:
Yes it's not ideal, that's because the `workloadmeta` does not provide a mock. I'd need to implement a small fake Store there.

Reply:
I've added the necessary piece `MockStore` in `workloadmeta`

Reply:
💯